scanner/mappers/mappers.go

```diff
@@ -592,83 +588,147 @@ var (
 
 // severityAndScores returns the severity and scores information out of a
 // ClairCore vulnerability. The returned information is dependent on the
-// underlying updater. If errors are found in the encoded information, partial
-// values might be returned with an error.
-func severityAndScores(vuln *claircore.Vulnerability, nvdVulns map[string]*nvdschema.CVEAPIJSON20CVEItem) (severityValues, error) {
-	var values severityValues
-	errList := errorhelpers.NewErrorList(fmt.Sprintf("%s: parsing severity and CVSS scores", vuln.Updater))
+// underlying updater.
+func severityAndScores(ctx context.Context, vuln *claircore.Vulnerability, nvdVulns map[string]*nvdschema.CVEAPIJSON20CVEItem) (severityValues, error) {
+	ctx = zlog.ContextWithValues(ctx,
+		"component", "mappers/severityAndScores",
+		"updater", vuln.Updater,
+		"vuln_id", vuln.ID,
+		"vuln_name", vuln.Name,
+	)
+
 	switch {
+	case rhelUpdaterPattern.MatchString(vuln.Updater):
+		return rhelSeverityAndScores(vuln)
 	case osvUpdaterPattern.MatchString(vuln.Updater):
-		if vuln.Severity == "" {
-			errList.AddStrings("severity is empty")
-		}
-		// ClairCore has no CVSS version indicator for OSV data, assuming CVSS V2 if
-		// not prefixed with a V3 version.
-		if strings.HasPrefix(vuln.Severity, "CVSS:3") {
-			if _, err := cvss3.VectorFromString(vuln.Severity); err != nil {
-				errList.AddError(err)
-			} else {
-				values.v3Vector = vuln.Severity
-			}
-		} else if _, err := cvss2.VectorFromString(vuln.Severity); err != nil {
-			errList.AddError(err)
-		} else {
-			values.v2Vector = vuln.Severity
+		sev, err := osvVectors(vuln)
+		if err != nil {
+			zlog.Debug(ctx).
+				Err(err).
+				Str("severity", vuln.Severity).
+				Msg("parsing severity, falling back to NVD")
+			break
 		}
-	case rhelUpdaterPattern.MatchString(vuln.Updater):
-		if vuln.Severity == "" {
-			errList.AddStrings("severity is empty")
+		return sev, nil
+	default:
```

Comment:
This is redundant, right?

Reply:
yeah, I can remove it. I sometimes add it and sometimes don't. Not sure why, honestly